North-Seattle-College/ad440-winter2021-thursday-repo
# Feature/task03/completeUserId/tasks_function

# Resolves Task/BUG :[#93](https://github.com/North-Seattle-College/ad440-winter2021-thursday-repo/issues/93)

# To Test API function users/{userId}/tasks 
- Test this function on Azure:
- API call - [users/2/tasks](https://nsc-functionsapp-team1.azurewebsites.net/api/users/2/tasks)
- This code completes the implementation of the Azure function for the previous PR #107  that is already merged into the dev branch
           - The following methods got implemented and Updated: GET/POST

- Once this PR gets approved the function will be deployed automatically into the development branch
-  To test this feature locally make sure to have the following: 
      - Azure function extension for VSCode
      - Pull the branch to your machine and make sure to be on the correct branch
      - Install the ODBC driver so you can connect to the database: For any issues with the ODBC driver checkout PR49 for additional documentation to fix ODBC driver
      -In local.settings.json at the root of the functions folder, enter in the values for the keys below in the following format: { "Values": { "KEY": "VALUE" } }
ENV_DATABASE_NAME (name of the database)
ENV_DATABASE_PASSWORD
ENV_DATABASE_SERVER (this is the server location)
ENV_DATABASE_USERNAME
- You can click F5 or click run->Start Debugging in the functions folder to start it up

- The GET request takes the userId and returns all a specific user by their Id. 
- The POST request is not allowed for this call.
- Error handling and Debug handling included in this feature





# time spent: 
I spent 10 hours completing this task.
